client/src/pages/Profile/Profile.tsx

```diff
@@ -2,43 +2,57 @@ import { Grid, List, ListItem, ListItemText, Paper } from "@material-ui/core";
 import useStyles from './useStyles';
 import { useState } from 'react';
 import MainPanel from '../../components/ProfileSettings/MainPanel/MainPanel';
+import { BrowserRouter as Router, Switch, Route, Link, useRouteMatch } from 'react-router-dom';
+import { v4 as uuid } from 'uuid';
 
 export default function Profile(): JSX.Element {
     const classes = useStyles();
 
+    const { path, url } = useRouteMatch();
     const [ compIndex, setIndex ] = useState(0);
 
     const listText = [
-        "Edit Profile",
-        "Profile photo",
-        "Availability",
-        "Payment",
-        "Security",
-        "Settings",
+        { link: 'editProfile', text:"Edit Profile" },
+        { link: 'profilePhoto', text: "Profile Photo" },
+        { link: 'availability', text: "Availability" },
+        { link: "payment", text: "Payment" },
+        { link: "security" , text: "Security" },
+        { link: "settings", text: "Settings" }
     ]
+    console.log(compIndex);
 
     return(
-        <Grid container className={classes.root} spacing={2}>
+        <Router>
+            <Grid container className={classes.root} spacing={2}>
             <Grid item xs={12}>
                 <Grid container>
                     <Grid key={1} item>
                         <List component="nav" className={classes.menuWrapper}>
-                            {listText.map((text, index) => {
+                            {listText.map((item, index) => {
                                 return(
-                                <ListItem selected={index === compIndex} key={index} button onClick={() => setIndex(index)}>
-                                    <ListItemText primary={text} />
-                                </ListItem>
+                                <Link to={`${url}/${item.link}`} key={uuid()} style={{textDecoration: 'none'}}>
+                                    <ListItem selected={index === compIndex} button onClick={() => setIndex(index)}>
```

Comment:
Since we are using react-router for routing, I think you can remove the onClick event for this ListItem button